lib/core/activity_tracker.dart

```diff
@@ -0,0 +1,14 @@
+class ActivityTracker {
+  bool _isToppingUp = false;
+  // getters
+  bool get isToppingUp => _isToppingUp;
+
+  // setters
+  void setToppingUp(bool value) {
```

Comment:
The setter should be defined as so:
```suggestion
  set setToppingUp(bool value) {
```